✨ [FEAT] CalendarController 생성 및 캘린더 기능 API 구현

wjd-hyeon opened the pull request:
## 🔘Part

- [x] 캘린더 기능 구현

  <br/>

## ➕ 이슈 링크

- #72

<br/>

## 🔎 작업 내용

- 캘린더 생성

- 캘린더 수정, 삭제, 검색

- 특정 날짜 일정 조회

- 키워드 검색

- 로그인한 부모의 캘린더 조회

  <br/>

wjd-hyeon committed:
🔨 refactor: CalendarController 응답 포맷을 CustomResponse로 통일

hambining commented on src/main/java/final_project/momeasy/domain/calendar/dto/CalendarResponseDto.java:
```diff
@@ -0,0 +1,25 @@
+                .content(calendar.getContent())
+                .build();
+    }
+}
```
이 부분은 Converter 클래스로 따로 분리하는 게 좋아보입니다 !
다른 도메인에서도 DTO-Entity 간 변환은 컨버터 안에서 하고 있거덩요

hambining commented on src/main/java/final_project/momeasy/domain/calendar/controller/CalendarController.java:
```diff
@@ -0,0 +1,77 @@
+            @RequestBody @Valid CalendarRequestDto requestDto,
+            @AuthenticationPrincipal CustomUserDetails userDetails) {
+        Long parentId = userDetails.getParent().getId();
+        return ResponseEntity.ok(calendarService.createCalendar(requestDto, parentId));
```
지금 프로젝트에서는 @AuthParent 라는 커스텀 애노테이션 사용하고 있어서 인가 부분은 다 통일 해주셔야 합니다요
그러면 Parent 객체 바로 넘겨서 서비스도 같이 수정하면 될 것 같습니다 !!
pr https://github.com/SMU-25/server/pull/54 참고하시면 될 것 같아용

hambining commented on src/main/java/final_project/momeasy/domain/calendar/service/CalendarService.java:
```diff
@@ -0,0 +1,111 @@
+                .title(requestDto.getTitle())
+                .content(requestDto.getContent())
+                .parent(parent)
+                .build();
```
요기 엔티티로 변환하는 부분도 컨버터에서 해야할 것 같습니다 🤓

hambining commented on src/main/java/final_project/momeasy/domain/calendar/controller/CalendarController.java:
```diff
@@ -0,0 +1,77 @@
+        return ResponseEntity.ok(calendarService.createCalendar(requestDto, parentId));
+    }
+
+    @GetMapping("/{id}")
```
현재 엔드포인트에서는 id만으로도 충분히 명확한 상황이라 괜찮은데 만약 추후에 childId나 parentId 같은 다른 것들이 섞일 수도 있으니까 calendarId처럼 명확하게 네이밍하는 것도 좋을 것 같습니다 🥸 어떻게 생각하시나요 알려주세요 !!!!!

hambining commented on src/main/java/final_project/momeasy/domain/calendar/controller/CalendarController.java:
```diff
@@ -0,0 +1,77 @@
+    private final CalendarService calendarService;
+
+    @PostMapping
+    public ResponseEntity<CalendarResponseDto> createCalendar(
```
응답 부분도 ResponseEntity 대신 CustomResponse 만들어서 사용하고 있어서 응답도 통일 부탁드립니다아-!!!

jh010303 replied:
@Operation을 통해 이 함수가 어떤 역할을 하는지 추가해주세요 !

hambining commented on src/main/java/final_project/momeasy/domain/calendar/service/CalendarService.java:
```diff
@@ -0,0 +1,111 @@
+    // 1. 일정 추가
+    public CalendarResponseDto createCalendar(CalendarRequestDto requestDto, Long parentId) {
+        Parent parent = parentRepository.findById(parentId)
+                .orElseThrow(() -> new EntityNotFoundException("해당 부모를 찾을 수 없습니다."));
```
여기도 커스텀 에러로 통일하는 게 더 좋을 것 같습니다
위에서 만든 `CalendarErrorCode`랑 `CustomException`으로 하면 될 것 같아용

jh010303 replied:
new CanendarException(CalendarErrorCode.~)로 오류 처리 해주시면 될 것 같습니다

hambining reviewed (changes requested):
고생하셨습니다-!!! 리뷰 남긴 거 한 번 읽어보시구 수정하시면 될 것 같습니다 !!!

**전체적으로 수정할 부분**
1. 응답 통일 (ResponseEntity -> CustomResponse, 에러 처리 -> CustomException & CustomErrorCode)
2. 엔티티 <-> DTO 간 변환 Converter 클래스 사용
3. 인가 부분 커스텀 애노테이션 (`@AuthenticationPrincipal` -> `@AuthParent`)

jh010303 commented on src/main/java/final_project/momeasy/domain/calendar/controller/CalendarController.java:
```diff
@@ -0,0 +1,77 @@
+    }
+
+    @DeleteMapping("/{id}")
+    public ResponseEntity<Void> deleteCalendar(
```
반환타입을 String으로 하여 "~ 삭제 완료" 응답을 하도록 추가해주세요 ! CustomResponse 사용하면 됩니다

jh010303 commented on src/main/java/final_project/momeasy/domain/calendar/entity/Calendar.java:
```diff
@@ -11,25 +11,33 @@
 @Builder
 @NoArgsConstructor(access = AccessLevel.PROTECTED)
-@AllArgsConstructor(access = AccessLevel.PRIVATE)
+@AllArgsConstructor // 기본 public 접근 제한자
```
entity를 외부에서 마음대로 생성할 수 있으니 @AllArgsConstructor(access = AccessLevel.PRIVATE)를 사용합시다 !

jh010303 commented on src/main/java/final_project/momeasy/domain/calendar/service/CalendarService.java:
조회는 CalendarQueryService에 생성, 삭제, 수정은 CalenarService로 나눠서 책임을 분리하는게 유지보수하기 좋아 보입니다 !

jh010303 commented on src/main/java/final_project/momeasy/domain/calendar/service/CalendarService.java:
```diff
@@ -0,0 +1,111 @@
+    }
+
+    // 4. 일정 수정 (💡 save 추가됨!)
+    public CalendarResponseDto updateCalendar(Long id, CalendarRequestDto requestDto, Long parentId) {
```
수정, 추가, 삭제 함수에 @Transactional 추가해주세요 !

jh010303 reviewed (changes requested):
고생하셨어요 !! 😁리뷰한거 한번 쭉 훑어보시기 바랍니다 !! 

**전체적으로 수정할 부분** 
@Operation 추가
@Transactional 추가
Entity에  @AllArgsConstructor(access = AccessLevel.PRIVATE)로 변경

wjd-hyeon committed:
🔨 refactor: CalendarController 응답 포맷을 CustomResponse로 통일

- 응답 객체 ResponseEntity → CustomResponse 로 변경
- Swagger용 @Operation(summary = "...") 추가

wjd-hyeon committed:
🔨 refactor: calendarId로 명칭 통일

wjd-hyeon committed:
🔨 Refactor: CalendarResponseDto 변환 로직을 Converter로 분리

wjd-hyeon committed:
🔨 Refactor: Calendar 엔티티 생성자 접근 제한 강화

wjd-hyeon committed:
🔨 refactor: CalendarController에 @AuthParent 적용

wjd-hyeon committed:
🔨 refactor: 메서드 파라미터를 Parent 객체 기반으로 수정

wjd-hyeon committed:
🔨 Parent ID 대신 Parent 객체로 조회하도록 수정

wjd-hyeon committed:
🔨 refactor: Parent ID 대신 객체 직접 사용하도록 통일

wjd-hyeon committed:
🔨 refactor: 예외 처리 CustomException 기반으로 통일

wjd-hyeon committed:
🔨 refactor: DTO-Entity 변환 로직을 CalendarConverter로 분리

wjd-hyeon committed:
🔨 refactor: 캘린더 삭제 응답을 CustomResponse<String> 형식으로 수정

wjd-hyeon committed:
🔨 refactor: CalendarService에서 조회 로직 분리하여 QueryService로 분리

wjd-hyeon committed:
🔨 refactor: CalendarController 조회 로직을 CalendarQueryService로 분리

wjd-hyeon committed:
🔨 refactor: 생성·수정·삭제 메서드에 @Transactional 추가

wjd-hyeon committed:
✨ feat: CalendarErrorCode에 예외 코드 추가

wjd-hyeon committed:
🔨 refactor: CalendarRepository의 날짜 조회 방식 개선 (LocalDate)

wjd-hyeon committed:
🔨 refactor: @AuthParent 적용하여 인가 방식 통일

wjd-hyeon committed:
✨ feat: 엔티티 변환 로직 Converter로 구현

wjd-hyeon committed:
🔨 refactor: Calendar 예외 처리 CustomException으로 통일

wjd-hyeon committed:
✨ feat: 조회 기능 CalendarQueryService로 분리

wjd-hyeon committed:
🔨 refactor: CalendarRepository의 날짜 조회 방식 개선

wjd-hyeon committed:
🔨 refactor: Calendar 예외 처리 CustomException으로 통일

wjd-hyeon committed:
✨ feat: Symptom 기능 및 API 구현

wjd-hyeon committed:
✨ feat: Illness 기능 및 API 구현

wjd-hyeon committed:
✨ feat: Illness 기능 및 API 구현

wjd-hyeon committed:
Merge remote-tracking branch 'origin/JeongHyeon' into JeongHyeon

# Conflicts:
#	src/main/java/final_project/momeasy/domain/calendar/service/CalendarService.java

wjd-hyeon committed:
Merge branch 'develop' into JeongHyeon

wjd-hyeon committed:
✨ feat: Illness 도메인 코드 리팩토링

wjd-hyeon committed:
✨ feat: IllnessResponse와 변수명 통일

wjd-hyeon committed:
✨ feat: Illness import문 리팩토링

wjd-hyeon committed:
✨ feat: Symptom import문 리팩토링

wjd-hyeon committed:
Merge remote-tracking branch 'origin/JeongHyeon' into JeongHyeon

wjd-hyeon committed:
🔨 refactor: 잘못된 에러 코드 상수명 수정

hambining commented on src/main/java/final_project/momeasy/domain/calendar/service/CalendarService.java:
```diff
@@ -18,13 +19,15 @@ public class CalendarService {
     private final CalendarRepository calendarRepository;
 
     // 1. 일정 추가
+    @Transactional
```
`@Transactional` 애노테이션 메서드마다 다는 것 보다는 클래스에 달아서 일괄 적용되게 해도 좋을 거 같아용

hambining reviewed (dismissed):
고생하셧씁니다-!!!!! 담부턴 pr 나눠서 올려쥬세요 ..... 🥺

wjd-hyeon committed:
🔨 refactor: 클래스에 @Transactional 적용

jh010303 reviewed (approved):
고생하셨습니다 !! 이제 깔끔한 pr 올려봅시다 !!
